app/destination/page.js

```diff
@@ -1,54 +1,60 @@
 "use client";
 
-import { useState } from 'react';
-
-import styles from '@/components/destination/destination.module.css';
-import { AddWishlistItem } from '@/components/destination/AddWishlistItem';
-
-// TASK - React 1 week 2
-// Move this to its own file
-const PlanetWishlistItem = ({
-  name,
-  onRemove,
-  thumbnail,
-}) => {
-  return (
-    <div className={styles.wishlistItem}>
-      <img className={styles.wishlistItemThumbnail} src={thumbnail} alt="" />
-      <b>{name.toUpperCase()}</b>
-      <button onClick={onRemove}>remove</button>
-    </div>
-  );
-}
-
+import { useState } from "react";
+import styles from "@/components/destination/destination.module.css";
+import { AddWishlistItem } from "@/components/destination/AddWishlistItem";
+import PlanetWishlistItem from "./PlanetWishlistItem";
+import { PlanetCard } from "./PlanetCard";
 
 export const Destinations = () => {
+  //state
   const [selectedPlanets, onAddPlanet] = useState([]);
-
-  let isPlanetSelected = false;
-  let numberOfPlanets = 0;
+  const [numberOfPlanets, setNumberOfPlanets] = useState(0);
+  const [isPlanetSelected, setIsPlanetSelected] = useState(false);
 
   const onAddOrRemovePlanet = (name, index) => {
-    // TASK - React 1 week 2
-    // Implement this function
-    // If you press the "ADD PLANET" the selected planet should display "SELECTED"
-    // And the counter should update, how many planets are selected (numberOfPlanets)
-    console.log(`You seleceted the following planet: ${name}, with the index of ${index}`);
-  }
+    if (selectedPlanets.includes(name)) {
+      const updatedPlanets = selectedPlanets.filter(
+        (planet) => planet !== name
+      );
+      onAddPlanet(updatedPlanets);
+
+      setNumberOfPlanets(updatedPlanets.length);
```

Comment:
This information can always come from the `selectedPlanets` as `selectedPlanets.length` since this number will update as the array grows :)